src/Service.Tests/dab-config.CosmosDb_NoSql.json

```diff
@@ -133,6 +142,63 @@
           ]
         }
       ]
+    },
+    "Earth": {
+      "source": "graphqldb.earth",
+      "permissions": [
+        {
+          "role": "anonymous",
+          "actions": [
+            {
+              "action": "create",
+              "fields": {
+                "include": [
+                  "id"
+                ],
+                "exclude": [
+                  "name"
+                ]
+              }
+            },
+            {
+              "action": "read",
+              "fields": {
+                "include": [
+                  "id"
+                ],
+                "exclude": [
+                  "name"
+      ]
+              }
+            },
+            "update",
+            "delete"
+          ]
+        },
+        {
+          "role": "authenticated",
+          "actions": [
+            "create",
+            {
+              "action": "read",
+              "fields": {
+                "include": [
+                  "*"
+                ],
+                "exclude": []
+              }
+            },
+            "update",
+            "delete"
+          ]
+    }
+      ],
+      "graphql": {
+        "type": {
+          "singular": "Earth",
```

Comment:
how about we try giving it a different singular name than the Entity name to make sure we test that scenario?

Reply:
This is a great point! I verified as below. 

1. Limit the field auth to authenticated. 
![image](https://user-images.githubusercontent.com/40870773/236576207-26a3a70c-c201-4c99-b925-c70186589339.png)
2. Grand the field auth to anonymous. 
![image](https://user-images.githubusercontent.com/40870773/236576273-03b1333d-983c-422f-b258-85ffb45da890.png)
3. Query field is correctly guarded by hot chocolate.
![image](https://user-images.githubusercontent.com/40870773/236575991-beec5c12-f44a-4b98-aa9a-0358b798b6ce.png)

Reply:
I'm not sure that I understand why would this be related to the changes in this PR?

Reply:
Since that will be helpful to test your function: `GetSchemaGraphQLFieldNamesForEntityName`

Reply:
Oh I see. Here is an existing test that has tested when entity name is not the same as the graph type. 
https://github.com/Azure/data-api-builder/blob/7efeec877b0ac53c3d40416096d03af43217fc05/src/Service.Tests/CosmosTests/QueryFilterTests.cs#LL624C11-L624C11
When ```Star``` entity name is ```StarAlias```
https://github.com/Azure/data-api-builder/blob/7efeec877b0ac53c3d40416096d03af43217fc05/src/Service.Tests/dab-config.CosmosDb_NoSql.json#L102

Reply:
Can we modify this existing test that uses alias names to add field level auth in the config file so that we can test your new functionality in the context of alias names? 

Particularly, Im concerned about this function `GetSchemaGraphQLFieldTypeByFieldName` when entity name is aliased. 

For this `StarAlias` example, the call in GraphQLFilterParser.Parse` will call this function with:

`GetSchemaGraphQLFieldTypeByFieldName('StarAlias', `name`)` 
and in the implementation of the function: we are looking for up field names based on `StarAlias` instead of `star` which may result in `Invalid filter object used as a nested field input value type` exception.
Can you please double check if Im missing something?

Reply:
Added a new test to exercise this scenario.